src/main/java/com/okta/tools/aws/settings/AwsINIConfiguration.java

```diff
@@ -0,0 +1,875 @@
+                    }
+                }
+                else
+                {
+                    writeProperty(out, node.getNodeName(), node.getValue(), separator);
+                }
+                first = false;
+            }
+            out.println();
+            out.flush();
+        }
+        finally
+        {
+            endRead();
+        }
+    }
+
+    /**
+     * Load the configuration from the given reader. Note that the
+     * {@code clear()} method is not called so the configuration read in will
+     * be merged with the current configuration.
+     *
+     * @param in the reader to read the configuration from.
+     * @throws ConfigurationException If an error occurs while reading the
+     *         configuration
+     * @throws IOException if an I/O error occurs
+     */
+    @Override
+    public void read(Reader in) throws ConfigurationException, IOException
+    {
+        BufferedReader bufferedReader = new BufferedReader(in);
+        Map<String, ImmutableNode.Builder> sectionBuilders = new LinkedHashMap<>();
+        ImmutableNode.Builder rootBuilder = new ImmutableNode.Builder();
+
+        createNodeBuilders(bufferedReader, rootBuilder, sectionBuilders);
+        ImmutableNode rootNode = createNewRootNode(rootBuilder, sectionBuilders);
+        addNodes(null, rootNode.getChildren());
+    }
+
+    /**
+     * Creates a new root node from the builders constructed while reading the
+     * configuration file.
+     *
+     * @param rootBuilder the builder for the top-level section
+     * @param sectionBuilders a map storing the section builders
+     * @return the root node of the newly created hierarchy
+     */
+    private static ImmutableNode createNewRootNode(
+            ImmutableNode.Builder rootBuilder,
+            Map<String, ImmutableNode.Builder> sectionBuilders)
+    {
+        for (Map.Entry<String, ImmutableNode.Builder> e : sectionBuilders
+                .entrySet())
+        {
+            rootBuilder.addChild(e.getValue().name(e.getKey()).create());
+        }
+        return rootBuilder.create();
+    }
+
+    /**
+     * Reads the content of an INI file from the passed in reader and creates a
+     * structure of builders for constructing the {@code ImmutableNode} objects
+     * representing the data.
+     *
+     * @param in the reader
+     * @param rootBuilder the builder for the top-level section
+     * @param sectionBuilders a map storing the section builders
+     * @throws IOException if an I/O error occurs
+     */
+    private void createNodeBuilders(BufferedReader in,
+                                    ImmutableNode.Builder rootBuilder,
+                                    Map<String, ImmutableNode.Builder> sectionBuilders)
+            throws IOException
+    {
+        ImmutableNode.Builder sectionBuilder = rootBuilder;
+        String line = in.readLine();
+        while (line != null)
+        {
+            line = StringUtils.stripEnd(line, null);
```

Comment:
➕ right trim instead of trim the entire line:
https://github.com/apache/commons-configuration/blob/trunk/src/main/java/org/apache/commons/configuration2/INIConfiguration.java#L433